### ISSUE TITLE
A file can't be dealt with on large-v2-ct model in Chinese.

### ISSUE THREAD

**Issue by satisl** [guillaumekln/faster-whisper#71]
Thanks for your work first. It's useful.
Howerver, there's still something wrong. It returns -1073740791 (0xC0000409) when dealing with a audio file in chinese. I have defined a function in which the variable 'result' is used to accept the 'segments' returned by the fast-whisper. It's normal in this function, but abnormal after being returned by the function. 
![image](https://user-images.githubusercontent.com/116792223/227228397-0bbe7da1-961e-431e-9db0-1ba64ee24880.png)
The line 'print(result)' works. 
![image](https://user-images.githubusercontent.com/116792223/227228957-62fa4f21-6edc-42d6-9f57-1f86a647965c.png)
But after the result is returned, python returns  -1073740791 (0xC0000409) and terminates
When changing the model or the language, it went  properly.
Confused.

**Comment by guillaumekln** [comment]
The same error is reported in #64 and is related to the cuDNN installation. Can you check that?

**Comment by satisl** [comment]
Really? But I have already installed Cudnn according to the Nivida documentation, and if I use medium-ct2 instead of large-v2-ct2, switch languages, or process other files, this type of problem will not occur.

**Comment by guillaumekln** [comment]
How much VRAM does your GPU have?

**Comment by satisl** [comment]
about 6000Mb and 4000mb when running

**Comment by guillaumekln** [comment]
Possibly you are running out of memory for this specific file. Can you try using `compute_type="int8_float16"`?

**Comment by satisl** [comment]
It still didn't work.

**Comment by guillaumekln** [comment]
Is it possible for you to share this audio file?

**Comment by satisl** [comment]
[the video file](https://1drv.ms/v/s!ArMIs28oZZeb-351nsKueh7MXmSD) 
with command
"ffmpeg -i "video file" -vn -ar 16000 "aac file"" 
"ffmpeg -i "video file" -vn -c:a libmp3lame -ar 16000 "mp3 file""
i extract the audio from the video

**Comment by guillaumekln** [comment]
I don't reproduce the error on Windows 11 with CUDA 11.8.0 and cuDNN 8.8.1.

Are you using the latest version of faster-whisper?

**Comment by satisl** [comment]
The first thing I do when this error occurred is to update the faster-whisper

**Comment by satisl** [comment]
Seems like that it's difficult to reproduce the error. Since this problem only occurs with this unique file under certain conditions (I have also processed various other files since then, and the result is normal operation). Perhaps this issue can be put on hold?

**Comment by satisl** [comment]
By coincidence, I discovered the cause of the error. When the model is unloaded, the program will crash.
Previously, model was a local variable and would automatically unload the model when the function ended. And the program crashes.
Now, model is a global variable and would automatically unload the model when the program ended.
![image](https://user-images.githubusercontent.com/116792223/227830483-85e4c2ec-4b00-466c-8d65-699a90df4553.png)
![image](https://user-images.githubusercontent.com/116792223/227830503-319d1e1f-5c83-4017-bd7e-c374fd7dd877.png)
Though the program will crash when unloading the model, it will happen after all things are finished now.
![image](https://user-images.githubusercontent.com/116792223/227830699-359b5c4e-4065-4ec0-a330-bfb91da85ffc.png)
![image](https://user-images.githubusercontent.com/116792223/227830718-2690435e-6b13-41cd-8864-05996843a02a.png)
Howerver, I have no idea why the program will crash when unloading the model.

**Comment by guillaumekln** [comment]
Does it also crash when you manually unload the model with `del model`?

**Comment by satisl** [comment]
No, it doesn't. It will when processing certain files (five out of approximately ninety files). If this error occurs when it processes a file, it will occur no matter how many times it processes the file.
My environment is rtx 3060 laptop, windows 10, cuda11.7, cudnn8.8.0, python3.10.10, model large-v2, language 'zh'.
Attempts have been made to reinstall the python environment or change the python version to 3.9.16, but this type of issue still exists.
The model have been redownloaded, but issue exits.

**Comment by satisl** [comment]
![image](https://user-images.githubusercontent.com/116792223/227942169-9b899acd-9748-4b52-ae1e-cefa82209076.png)
If the model is manually unloaded after processing the file, the program will crash
![image](https://user-images.githubusercontent.com/116792223/227942599-605c6e9a-8d18-4b21-9e28-21e76e48c275.png)

**Comment by ProjectEGU** [comment]
I found a way to run the transcription in a separate process so that even though it exits that child process, it doesn't exit your main script. Here is a working example:

```
from multiprocessing import Process, Manager
from faster_whisper import WhisperModel

def work_log(argsList, returnList):
    model_size = "large-v2"
    model = WhisperModel(model_size, device="cuda", compute_type="float16")
    segments, info = model.transcribe(*argsList, beam_size=5)
    returnList[0] = [list(segments), info]

# workaround to deal with a termination issue: https://github.com/guillaumekln/faster-whisper/issues/71
def runWhisperSeperateProc(*args):
    with Manager() as manager:
        returnList = manager.list([None])
        p = Process(target=work_log, args=[args, returnList])  # add return target to end of args list
        p.start()
        p.join()
        p.close()
        return returnList[0]

if __name__ == '__main__':
    segments, info = runWhisperSeperateProc("audio.mp3")
    print(segments, info)

```

**Comment by yslion** [comment]
same issues

**Comment by guillaumekln** [comment]
@ProjectEGU @yslion @DoodleBears Are you all using the library on Windows?

**Comment by guillaumekln** [comment]
I'm can now reproduce the issue on Windows.

It is somehow related to the temperature fallback. Can you try setting `temperature=0`?

**Comment by satisl** [comment]
Glad to know that the reason has been detected. With this setting, the program runs nomally. 
However, maybe it will produce slightly worsen result? I don't know.

**Comment by DoodleBears** [comment]
> @ProjectEGU @yslion @DoodleBears Are you all using the library on Windows?

Yes I using the library on Windows 10

**Comment by guillaumekln** [comment]
I have a possible fix for this issue in https://github.com/OpenNMT/CTranslate2/pull/1201, but I can't test on my Windows machine today. Can you help testing?

1. Go to the [build page](https://github.com/OpenNMT/CTranslate2/actions/runs/4829849143)
2. Download the artifact "python-wheels"
3. Extract the archive
4. Install the Windows wheel matching your Python version with `pip install --force-reinstall <wheel file>`

**Comment by DoodleBears** [comment]
> I have a possible fix for this issue in [OpenNMT/CTranslate2#1201](https://github.com/OpenNMT/CTranslate2/pull/1201), but I can't test on my Windows machine today. Can you help testing?
> 
> 1. Go to the [build page](https://github.com/OpenNMT/CTranslate2/actions/runs/4829849143)
> 2. Download the artifact "python-wheels"
> 3. Extract the archive
> 4. Install the Windows wheel matching your Python version with `pip install --force-reinstall <wheel file>`

Yes, I will try it now, by the way I try `temperature=0`, it works (process did not exit)

**Comment by DoodleBears** [comment]
when using `temperature=0`:
I met `segmentation fault` **BEFORE installing the wheel** sometimes when I call the function below many times, don't know why

```python
def transcribe_speeches(self):
    log.init_logging(debug=True)
    # NOTE: 读取音频文件
    logger.info(f"开始语音转文字")
    whisper = WhisperModel(WHISPER_MODEL, device="cuda", compute_type="float16")
    speeches_num = len(self.speeches)
    for index, speech in enumerate(self.speeches):
        logger.debug(f"开始识别 {speech.audio_path}")
        speech_text = ''
        # NOTE: 识别音频文件
        segments, _ = whisper.transcribe(
            audio=speech.audio_path,
            language='zh',
            vad_filter=False,
            temperature=0,
            initial_prompt='以下是普通话的句子。'
            )
        segments = list(segments)
        if len(segments) == 0:
            logger.warning(f"识别结果为空: {speech.audio_path}")
        else:
            speech_text = '，'.join([segment.text for segment in segments])
            logger.info(f"识别结果({index+1}/{speeches_num}): {speech_text}")
        self.speeches[index].text = speech_text
        
    
    logger.info(f"结束语音转文字: {self.speeches}")
    # queue.put(self.speeches)
    # FIXME: 卸载模型后会导致程序终止
    del whisper
```

**Comment by satisl** [comment]
`import os
import subprocess
from faster_whisper import WhisperModel

files = os.listdir('input')
for file in files:
    file = file.rsplit('.', 1)[0]
    print(file)
    command = f'ffmpeg -i \"input\\{file}.mp4\" -vn -ar 16000 -c:a libmp3lame -y \"tmp\\{file}.mp3\"'
    subprocess.run(command, capture_output=True, check=True)
    model = WhisperModel('large-v2', device='cuda', compute_type='float16')
    segment, info = model.transcribe(f'tmp\\{file}.mp3', language=None)
    for i in segment:
        text = i.text
    del model

print('finished')`
same error

**Comment by guillaumekln** [comment]
Thank you for the test! I will keep looking...

**Comment by fquirin** [comment]
I'm not sure if this is directly related, but I get `Segmentation fault` error from time to time when I start to analyze the transcription segments via `for segment in segments: ...`. I can't really pin down the precise location but it must be somewhere in `WhisperModel.generate_segments` and it happens only when my program tries to handle some remaining chunks at the end of a stream that are basically background noise.
Since I set `temperature=0` it hasn't happened again.

**Comment by guillaumekln** [comment]
@fquirin Are you also running on Windows with a GPU? If not, I’m not sure your issue is related. You can open another issue if you can share the audio and options triggering the crash.

**Comment by fquirin** [comment]
@guillaumekln I'm running it on Windows + CPU.
The problem is I can't reproduce it with audio files so far, only with my live-streaming server, but a pretty reliable way to get the segmentation fault is coughing 🤔. First I thought it was a problem with my code but it never happens with `temperature=0` and so far it never happend on Linux Aarch64 as well (with or without temp=0). Notably another difference to my Linux Aarch64 system is that my x86 CPU is much much faster (maybe a race condition?).

Btw, when I run my "coughing" test WAV files I noticed that Whisper can start to hallucinate pretty extensively with temperature != 0.

I'll try to pin down the segmentation fault by adding some debug info to `WhisperModel.generate_segments`

**Comment by Keith-Hon** [comment]
I have the same error when running the script in windows 10 WSL (ubuntu)

**Comment by archive-r** [comment]
same issue in windows 11

**Comment by archive-r** [comment]
I was able to avoid that error with the temperature=0 setting. Will this setting adversely affect the transcribe results? I searched the whisper repo, but couldn't find a satisfactory answer.

**Comment by zh-plus** [comment]
Same issue persists in Windows 11 with Python 3.10, CUDA 11.8, and CTranslate2 version 3.13.0. Even when setting the `temperature=0`, it still doesn't work. Are there any other possible workarounds available?

**Comment by guillaumekln** [comment]
Yes disabling the temperature fallback can affect the results. The fallback is mostly useful to recover from cases where the model generates the same token in a loop.

**Comment by archive-r** [comment]
> Yes disabling the temperature fallback can affect the results. The fallback is mostly useful to recover from cases where the model generates the same token in a loop.

Thank you. My test results were the same as you said.

**Comment by zh-plus** [comment]
> My runtime environment is Python 3.11.4, CUDA 11.8.0, graphics card driver 522.06, and cudnn-windows-x86_64-8.9.3.28. I am using the faster-whisper project, and when I try to load the model using GPU, Python returns -1073740791 (0xC0000409) error. However, when I use CPU, the error does not occur.
> 
> I have tried various solutions, including the ones you mentioned above, such as installing CUDA environment, adding system variables, and modifying the temperature to 0. None of them have worked.
> 
> Whenever I iterate over the segments, CUDA crashes, and the program terminates.
> 
> Finally, when I test and print print(torch.cuda.is_available()) to check if CUDA device is recognized as True, the program runs without any issues.
> 
> My personal estimation is that there might be an issue with the initialization and release of CUDA in CT2.

Go check if you install zlib refer to https://docs.nvidia.com/deeplearning/cudnn/install-guide/index.html#install-zlib-windows

**Comment by archive-r** [comment]
> Finally, when I test and print print(torch.cuda.is_available()) to check if CUDA device is recognized as True, the program runs without any issues.

If I understand correctly, is simply checking it (torch.cuda.is_available()) going to solve the problem? If so, that's interesting.

**Comment by guillaumekln** [comment]
@JamePeng This is a different issue. The issue described in this thread is a crash when unloading the model.

The error you get generally means that the program cannot locate the cuDNN and/or Zlib libraries. There are already several discussions about this.

**Comment by JamePeng** [comment]
@guillaumekln ok, it worked now, thanks for your help

**Comment by zh-plus** [comment]
Do we have any updates on resolving this issue? Currently, using the workaround of setting `temperature=0` is an option, but it could potentially impact the model's performance.

**Comment by CheshireCC** [comment]
> Does it also crash when you manually unload the model with `del model`?

I compileted my app with nuitka, and then run it as Administrastor User , it will not crash when unload model.

**Comment by sanek11591** [comment]
I have the same problem. My config python 3.10.7 CUDA ToolKit 11.8 cuDNN 8.9.6 and add to PATH.

**Comment by Dadangdut33** [comment]
i had the same problem and i think i fixed it in my case by moving the faster whisper import inside the function that needs/uses it.

**But,** keep in mind that **I am using faster whisper through stable whisper**, and i need to import some stuff from the faster whisper library. I previously imported it globally in the top and found that my app will sometimes crashes after loading and reloading different model, but then after moving it to only inside the function that uses it somehow the crash is gone

**Comment by 1Wayne1** [comment]
I have the same issue. I reinstall pytorch with this command `conda install pytorch==2.0.1 torchvision==0.15.2 torchaudio==2.0.2 pytorch-cuda=11.7 -c pytorch -c nvidia` and solve the problem.

**Comment by nebehr** [comment]
I can consistently reproduce it with the latest master, Python 3.11.1 and Cuda 12.5 on Windows 10, 3 minutes of audio and tiny model, with the following simple code:

```python
from faster_whisper import WhisperModel
model = WhisperModel("tiny", device="cuda", compute_type="auto")
segments, info = model.transcribe("js.wav")
for segment in segments:
    print("[%.2fs -> %.2fs] %s" % (segment.start, segment.end, segment.text))
```
I only installed Cuda Toolkit 12.5, not cuDNN. At no point does the system max out on CPU, GPU or memory.

If device="cpu" is forced, the issue does not occur, nor does it with temperature=0.0 as stated above. Curiously, it also does not occur if I don't iterate to the end of `segments` generator: in my case, if the iteration is stopped roughly half way through, there is no crash.

If I put `del model` at the end, on some occasions the crash comes on that instruction, but sometimes _after_ it.

**Comment by jianchang512** [comment]
Windows 10 using only cpu, data type int8 / float32, both may crash. Same feedback as above.


Possible reproductions: Multiple larger audios (10+ minutes/16k/ac 1/wav), for loop continuous recognition, in the last few audio file tasks, at the end of the segments iteration, regardless of whether del model or not, and regardless of whether multiple recognition tasks share a single model, or each task creates a single model, all may crash.


Temperature has been set to 0, condition_on_previous_text has been set to false, beam_size best_of has been set to 1

Single task execution, even with large audio, rarely crashes. Crashes mostly occur when multiple tasks in a row continue.

Tried creating a process for each task, and when one process finishes and then starts another, it still crashes!

Looking at the dump crash info for windows.

`The thread tried to read from or write to a virtual address for which it does not have the appropriate access.`

![image](https://github.com/user-attachments/assets/9fb31b12-3239-466c-a042-d66ecae00635)

**Comment by TechInterMezzo** [comment]
Did anyone find out yet if this is a bug in faster-whisper or in ctranslate2?

**Comment by usernotnull** [comment]
https://github.com/SYSTRAN/faster-whisper/issues/71#issuecomment-1526144389

This workaround worked for me as well.

Windows 11, Python 3.13.0, CUDA 12.4

**Comment by jeasonzhang-eth** [comment]
https://github.com/SYSTRAN/faster-whisper/issues/71#issuecomment-1526144389
work for me

**Comment by anthonyalayo** [comment]
I'm currently hitting this as well -- looking forward to the fix.

**Comment by Insolent-M1nx** [comment]
Any update on this. Seems to occur for me when trying to transcribe multiple files in sequence. This happens while launching as a subprocess (mentioned in #71) and with Temperature set to 0.

**Comment by dbarkar** [comment]
Happens to me every time I use faster_whisper model, I can run it only once per process. 
Every other whisper variations work fine.

**Comment by CoreAccess** [comment]
What worked for me is to add temperature=0.0 in the transcribe function like this

segments, _ = model.transcribe(
                audio_path,
                beam_size=5,
                language="en",
                temperature=0.0,
                condition_on_previous_text=False,
                vad_filter=True,
                vad_parameters=dict(min_silence_duration_ms=500),
                word_timestamps=True
            )

If I omit temperature, it will crash when the function ends or if I run "del model" it will also crash but when I put in the temperature argument, it doesn't fail and script execution continues as normal.

Zero clue as to why this is happening though, but man what an annoying bug because there are no errors for it being displayed anywhere and I cant get it even produce an error message, it just kills script execution, like magic lol.

**Comment by emcodem** [comment]
Also seen here:

https://github.com/Purfview/whisper-standalone-win/issues/381

and here, this also contains a minimal audio snippet to reproduce the behaviour, in my tests it appeared to be only reproduceable with certain audio content.
https://github.com/Purfview/whisper-standalone-win/issues/402

**Comment by baiyun2020** [comment]
Is there an updated solution to this issue? On Windows 10, it crashes when using 'del mode'. But setting temperature=0 works fine.

**Comment by Purfview** [comment]
>That is, until today! The workaround I've found is to switch out the model before unloading it.

It's confirmed many times that "del" doesn't help.
In your script you don't watch if temperature was above 0, you need to enable DEBUG logging to see it.

>This prevents any crashes on my system and allows me to unload models

Unloading model doesn't produce crash. Crash is produced on Python exit.
I wonder if it crash on latest's Python versions 3.14 - 3.15.

**Comment by ia319** [comment]
This appears to be the same Windows CUDA model cleanup issue, but I isolated one additional factor in my environment.

In my app, the crash happened when replacing a loaded faster-whisper CUDA model with another model in the same worker process. The transcription completed successfully, but the process aborted when the old CTranslate2 Whisper object was released.

I reproduced the behavior with a minimal script:

- default allocator / `cuda_malloc_async`: crashes when the CUDA model is destroyed
- `CT2_CUDA_ALLOCATOR=cub_caching`: `base -> release -> small -> release` completes successfully

My current workaround is to set this before importing faster-whisper/CTranslate2 on Windows:

```python
os.environ.setdefault("CT2_CUDA_ALLOCATOR", "cub_caching")
```

This keeps user overrides intact, but avoids the default async CUDA allocator path that crashes during model cleanup in my environment.
